### ISSUE TITLE
Refactor OrbitFinder

### ISSUE THREAD

**Issue by thkruz** [thkruz/keeptrack.space#804]
### User Story
- **As a** developer of the Keep Track project,
- **I want** to refactor the `OrbitFinder` class to reduce its complexity and improve logical flow,
- **So that** the codebase becomes more maintainable, and methods within the class are more encapsulated and less inter-dependent.

### Acceptance Criteria
- **Given** the current state of the `OrbitFinder` class with high complexity and interdependent methods,
- **When** I refactor the class,
- **Then** the class should exhibit lower complexity, have a more logical flow, and its methods should be more encapsulated and less inter-dependent.

### Tasks
- [ ] Analyze the current `OrbitFinder` class to identify areas of high complexity and inter-dependency.
- [ ] Design a refactoring plan that outlines the steps to reduce complexity and improve logical flow.
- [ ] Refactor methods to be private where appropriate, ensuring encapsulation and reducing unnecessary exposure of class internals.
- [ ] Implement changes to reduce the inter-dependency of methods within the class.
- [ ] Conduct code reviews and testing to ensure the refactored class maintains its functionality and adheres to project standards.

### Commentary
The refactoring of the `OrbitFinder` class is crucial for the long-term maintainability of the Keep Track codebase. This task should focus on improving code readability and reducing the complexity without altering the external behavior of the class. Collaboration and continuous integration practices should be followed to ensure a smooth transition and effective refactoring process.

**Comment by thkruz** [comment]
## Status update

The `OrbitFinder` this issue targeted no longer exists in its old form. It was deleted from `src/app/analysis/orbit-finder.ts`, rewritten, and moved into the ootk submodule at `src/engine/ootk/src/orbit-design/OrbitFinder.ts` (commit `45332cc`). Consumers (`breakup`, `edit-sat`, `new-launch`) now import it from `@ootk/src/main`.

Some of the original tasks are already done, but the complexity and coupling problems remain.

### Already done

- [x] Methods are private where appropriate. Every method except the public entry point `rotateOrbitToLatLon()` uses the `_` suffix.
- [x] Now lives in ootk as pure math, with no singleton or DOM coupling.
- [x] A test file exists (`OrbitFinder.test.ts`), but it only covers TLE-string formatting.

### Still open

1. `rotateOrbitToLatLon()` is a ~115-line method with two largely-parallel branches (elliptical vs circular) and all orchestration inlined.
2. Methods are coupled through shared mutable state: `currentParams_`, `lastLatitude_`, and `currentDirection_` are mutated as side effects across `meanACalc_`, `updateOrbit_`, `determineDirection_`, and `meanACalcLoop_`. The `successfulDirection` save/restore is a symptom of this.
3. Propagation logic is duplicated. The sequence (generate TLE, `createSatrec`, `calculateTimeVariables_`, `propagate`, `eci2lla`) is repeated in 5 methods. `updateOrbit_` and `updateOrbitWithoutDirectionCheck_` are nearly identical, differing only in the direction check.

### Blocker

The existing test only pins TLE-string formatting, not the search. There is no coverage to confirm the refactor preserves behavior, which is the main requirement here.

### Tasks

- [ ] Add tests for the search logic first (`rotateOrbitToLatLon` end-to-end: circular and elliptical, N/S direction, RAAN/longitude convergence). These need to land before any refactor.
- [ ] Extract a private `propagateAt_(params)` returning `{ lat, lon, alt }` to remove the duplication.
- [ ] Fold `updateOrbitWithoutDirectionCheck_` into `updateOrbit_` with a `checkDirection` flag.
- [ ] Split `rotateOrbitToLatLon()` into `rotateElliptical_()` and `rotateCircular_()`.
- [ ] Reduce the shared mutable state so the direction save/restore is no longer needed.

Keep open, but rescope to the above. The encapsulation and relocation work is complete. The refactor now lands in the ootk submodule, not the main repo.

**Comment by thkruz** [comment]
Replaced the entire class with an improved rv2tle method that reduced error on the new TLEs from ~3km to meters.
